src/proxies/forwarded.rs

```diff
@@ -0,0 +1,647 @@
+use crate::{
+    headers::{HeaderName, HeaderValue, Headers, ToHeaderValues, FORWARDED},
+    parse_utils::{parse_quoted_string, parse_token},
+};
+use std::{borrow::Cow, convert::TryFrom, fmt::Write, net::IpAddr};
+
+// these constants are private because they are nonstandard
+const X_FORWARDED_FOR: HeaderName = HeaderName::from_lowercase_str("x-forwarded-for");
+const X_FORWARDED_PROTO: HeaderName = HeaderName::from_lowercase_str("x-forwarded-proto");
+const X_FORWARDED_BY: HeaderName = HeaderName::from_lowercase_str("x-forwarded-by");
+
+/// A rust representation of the [forwarded
+/// header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded).
+#[derive(Debug, Clone, Default)]
+pub struct Forwarded<'a> {
+    by: Option<Cow<'a, str>>,
```

Comment:
Necessary, I believe. I'll try it to be sure, though

Reply:
If this were '_, I believe there would be no assurance to the compiler that the forwarded has the same lifetime as the `&impl AsRef<Headers>`, and that all the Cows have same lifetime as the request/response. And we need that because in the borrowed state, we're just pointing at data held in the request/response

Reply:
Ok, I really don't understand lifetimes in structures yet, I think.